utils/videoFileParser/video_es_parser.cpp

```diff
@@ -0,0 +1,1159 @@
+/*
+Copyright (c) 2023 - 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <string.h>
+#include "video_es_parser.h"
+#include "../../src/parser/hevc_defines.h"
```

Comment:
as we discussed, we need to expose another api for this and I don't think it will fit within the existing parser api

Reply:
Will remove the reference to the internal headers. 

Note this is a different parser. Think of it as a simple demuxer, which parses the files of AVC/HEVC/AV1 elementary stream files and get the picture data, and it can also read IVF container files. It is in parallel with roc_video_dec, ffmpeg demuxer in utility. The interface is exposes the same way as roc_video_dec, and ffmpeg demuxer. It is another utility.